src/LDAP/Search/Filter/Binary.hs

```diff
@@ -121,20 +120,24 @@ leadKeychar = alpha
 keychar :: Get Char
 keychar = asum [ alpha, digit, hyphen ]
 
-utf1Subset :: Get Char
-utf1Subset = Char8.satisfy $ not . (`elem` ['\NUL', '(', ')', '*', '\ESC'])  -- fixme: this parses 129-255 too
-
+-- Idea: the escaped portion of the valueEncoding and the subset exceptions
+-- only apply to ASCII code points. We can thus process it first without any
+-- concern for multi-byte encoding, then process the UTF-8 byte string as a
+-- whole.
 valueEncoding :: Get Text
-valueEncoding = Text.pack <$> many (normal <|> escaped)
+valueEncoding = toText <$> many valueEncodingByte
     where
-        normal = utf1Subset
+        -- 8-bit chars to Text. TODO: handle decoding error
+        toText = Text.pack . decodeString
+        valueEncodingByte :: Get Char
+        valueEncodingByte = escaped <|> normal
+        -- TODO? make excluded characters trigger a parse error by flipping the satisfy check around and failing explicitly
```

Comment:
Is this suggestion for the sake of error reporting? What error does `satisfy` produce?